api/components/com_media/src/Controller/MediaController.php

```diff
@@ -0,0 +1,401 @@
+	 *
+	 * @var    string
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $contentType = 'media';
+
+	/**
+	 * Query parameters => model state mappings
+	 *
+	 * @var  array
+	 */
+	private static $listQueryModelStateMap = [
+		'path'    => [
+			'name' => 'path',
+			'type' => 'STRING'
+		],
+		'url'     => [
+			'name' => 'url',
+			'type' => 'BOOLEAN'
+		],
+		'temp'    => [
+			'name' => 'temp',
+			'type' => 'BOOLEAN'
+		],
+		'content' => [
+			'name' => 'content',
+			'type' => 'BOOLEAN'
+		],
+	];
+
+	private static $itemQueryModelStateMap = [
+		'path'    => [
+			'name' => 'path',
+			'type' => 'STRING'
+		],
+		'url'     => [
+			'name' => 'url',
+			'type' => 'BOOLEAN'
+		],
+		'temp'    => [
+			'name' => 'temp',
+			'type' => 'BOOLEAN'
+		],
+		'content' => [
+			'name' => 'content',
+			'type' => 'BOOLEAN'
+		],
+	];
+
+	/**
+	 * The default view for the display method.
+	 *
+	 * @var    string
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $default_view = 'media';
+
+	/**
+	 * Display a list of files and/or folders.
+	 *
+	 * @return  static  A \JControllerLegacy object to support chaining.
+	 *
+	 * @throws  \Exception
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function displayList()
+	{
+		// Set list specific request parameters in model state.
+		$this->setModelState(self::$listQueryModelStateMap);
+
+		// Display files in specific path.
+		if ($this->input->exists('path'))
+		{
+			$this->modelState->set('path', $this->input->get('path', '', 'STRING'));
+		}
+
+		// Return files (not folders) as url's.
+		if ($this->input->exists('url'))
+		{
+			$this->modelState->set('url', $this->input->get('url', true, 'BOOLEAN'));
+		}
+
+		// Map JSON:API compliant filter[search] to com_media model state.
+		$apiFilterInfo = $this->input->get('filter', [], 'array');
+		$filter        = InputFilter::getInstance();
+
+		// Search for files matching (part of) a name or glob pattern.
+		if ($doSearch = array_key_exists('search', $apiFilterInfo))
+		{
+			$this->modelState->set('search', $filter->clean($apiFilterInfo['search'], 'STRING'));
+			// Tell model to search recursively
+			$this->modelState->set('search_recursive', $this->input->get('search_recursive', false, 'BOOLEAN'));
+		}
+
+		return parent::displayList();
+	}
+
+	/**
+	 * Display one specific file or folder.
+	 *
+	 * @param   string  $path  The path of the file to display. Leave empty if you want to retrieve data from the request.
+	 *
+	 * @return  static  A \JControllerLegacy object to support chaining.
+	 *
+	 * @throws  InvalidPathException
+	 * @throws  \Exception
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function displayItem($path = '')
+	{
+		// Set list specific request parameters in model state.
+		$this->setModelState(self::$itemQueryModelStateMap);
+
+		// Display files in specific path.
+		$this->modelState->set('path', $path ?: $this->input->get('path', '', 'STRING'));
+
+		// Return files (not folders) as url's.
```

Comment:
plural, not ownership, correct english

```suggestion
		// Return files (not folders) as urls.
```